openedx/core/djangoapps/credit/api/eligibility.py

```diff
@@ -266,7 +267,9 @@ def set_credit_requirement_status(username, course_key, req_namespace, req_name,
     # If we're marking this requirement as "satisfied", there's a chance
     # that the user has met all eligibility requirements.
     if status == "satisfied":
-        CreditEligibility.update_eligibility(reqs, username, course_key)
+        is_eligible, eligibility_record_created = CreditEligibility.update_eligibility(reqs, username, course_key)
+        if eligibility_record_created and is_eligible:
+            send_credit_notifications(username, course_key)
```

Comment:
Please wrap this in a try/except.  If the email for some reason fails to send, we don't want to roll back the transaction!